shub/deploy.py

```diff
@@ -75,7 +75,7 @@
 @click.option("-k", "--keep-log", help="keep the deploy log", is_flag=True)
 def cli(target, version, debug, egg, build_egg, verbose, keep_log):
     conf, image = load_shub_config(), None
-    if target in conf.projects:
+    if target != 'default':
```

Comment:
I guess it's a piece of my recent changes: the reason is that there's a case when getting target config is not needed at all, i.e. when user wants to build a project egg only with `--build-egg` [flag](https://github.com/scrapinghub/shub/blob/master/shub/deploy.py#L95). The deploy module code is "branchy" now, probably we could transform this build-egg case to click eager option with callback?